### PR TITLE
create calc, plus needed routers and controllers

### PR DESCRIPTION
**Pull request by Can-Tina**

### CODE REVIEW

File: src/client/components/calcAvg.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+
+const [strengthAvg, setStrengthAvg] = useState(0)
+const [companionAvg, setCompanionAvg] = useState(0)
+const [designAvg, setDesignAvg] = useState(0)
+const [ratingsArray, setRatingsArray] = useState([])
+
+const fetchRatings = () => {
+    const ratingUrl = "http://localhost:4000/ratings"
```
Comment (nathanSamosa) [inline src/client/components/calcAvg.js]: now that Theunis has implemented the env url we should go back and make sure we havent missed any hardcoded routes

Comment (TheunisK) [inline src/client/components/calcAvg.js]: We will need to change this to process.env.REACT_APP also.

File: src/server/routers/rating.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    postRating
+} = require("../controllers/rating");
+
+const router = express.Router()
+
+router.get("/", getRatings)
+
+router.post("/", postRating)
+
+module.exports = router;
```
Comment (TheunisK) [inline src/server/routers/rating.js:13]: Other than the URL being visible this looks great!

File: src/client/components/calcAvg.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    const calculateDesign = () => {
+        let designTotal = 0;
+        for(let i = 0; i < ratingsArray.length; i++) {
+            designTotal = designTotal + ratingsArray[i].design
+        }
+        let designAverage = designTotal/ratingsArray.length
+        setDesignAvg(designAverage)
+    }
+}
```
Comment (nathanSamosa) [inline src/client/components/calcAvg.js:42]: very clean calculation! we could possibly look at a way to combine these into a single function with a ratings key parameter.

File: src/server/routers/rating.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+const {
+    getRatings,
+    postRating
+} = require("../controllers/rating");
+
+const router = express.Router()
+
+router.get("/", getRatings)
+
+router.post("/", postRating)
```
Comment (nathanSamosa) [inline src/server/routers/rating.js:11]: love the multi-use route

File: src/client/components/calcAvg.js (hunk 1 above)
Comment (Can-Tina) [reply src/client/components/calcAvg.js]: Yes, can do that now

Comment (Can-Tina) [reply src/client/components/calcAvg.js]: Done

File: src/server/routers/rating.js (hunk 2 above)
Comment (Can-Tina) [reply src/server/routers/rating.js:13]: Thanks!

File: src/client/components/calcAvg.js (hunk 1 above)
Comment (Can-Tina) [reply src/client/components/calcAvg.js]: Good plan

File: src/client/components/calcAvg.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+const url = process.env.REACT_APP_API_URL
```
Comment (TheunisK) [inline src/client/components/calcAvg.js:2]: Much better!

File: src/client/components/calcAvg.js (hunk 1 above)
Comment (glowkeeper) [reply src/client/components/calcAvg.js]: 👍🏿

**Comment by glowkeeper** [comment]
Approved

**Pull request merged**